app/api/entities/metadataExtraction/saveSelections.ts

```diff
@@ -0,0 +1,87 @@
+import { files } from 'api/files';
+import { uniqBy } from 'lodash';
+import { ExtractedMetadataSchema } from 'shared/types/commonTypes';
+import { EntitySchema } from 'shared/types/entityType';
+import { FileType } from 'shared/types/fileType';
+import { textSimilarityCheck } from './textSimilarityCheck';
+import { ISO6391toISO6392 } from '../../../shared/languagesList';
+
+interface EntityWithExtractedMetadata extends EntitySchema {
+  __extractedMetadata: { selections: ExtractedMetadataSchema[] };
+}
+
+const updateSelections = (newSelections: any[], storedSelections: any[]) => {
+  const merged = newSelections.concat(storedSelections);
+  const selections = uniqBy(merged, 'name');
+  return selections;
+};
+
+const removeUserChangedSelections = (entityData: { [key: string]: any }, userSelections: any[]) => {
+  let updatedSelections = [];
+
+  if (userSelections.length > 0) {
+    updatedSelections = userSelections.filter(selection => {
+      if (selection.name === 'title') {
+        return textSimilarityCheck(selection.selection.text, entityData.title);
+      }
+      return textSimilarityCheck(
+        selection.selection.text,
+        entityData[selection.name][0].value || ''
+      );
+    });
+  }
+
+  return updatedSelections;
+};
+
+const prepareSelections = (entity: EntityWithExtractedMetadata, file: FileType) => {
+  let selections = entity.__extractedMetadata?.selections || [];
+
+  const entityData = {
+    title: entity.title,
```

Comment:
This might give problems if some database has a template with a field called `Title`. Idk if it's worth fixing.